lib/datadog/appsec/contrib/configuration/settings.rb

```diff
@@ -1,20 +0,0 @@
-# frozen_string_literal: true
-
-require_relative '../../../core/configuration/base'
-
-module Datadog
-  module AppSec
-    module Contrib
-      module Configuration
-        # Common settings for all integrations
-        # TODO: move to datadog/appsec/configuration or remove?
-        class Settings
-          include Datadog::Core::Configuration::Base
-
-          option :enabled, default: true
-          option :service_name
```

Comment:
Can the appsec instrumentations not be disabled/use the service name?

Reply:
Not at the moment https://docs.datadoghq.com/security/application_security/enabling/ruby/?tab=rails#get-started

Reply:
Since AppSec now uses Tracing spans as transport, AppSec has no way to actually define, use, or report service names, so no for the service name.

Disabling could theoretically be useful, but would largely be equivalent to commenting out, with the difference that it is evaluated so would be runtime-checked.

The only cases that would be remotely useful would be:

- forcefully disabling Rack when it is added automatically by Rails, but that's not supported anyway (AppSec would not work, it _needs_ the Rack integration), but you get the theory.
- disabling select instrumentations when auto-instrumenting is in use (e.g disabling Rails and/or Sinatra in a Rails app with nested Sinatra ones, leaving only Rack or something)

Reply:
Thanks for the extra context @lloeki